src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/HMACHashProvider.Browser.Native.cs

```diff
@@ -43,42 +43,35 @@ public override int GetCurrentHash(Span<byte> destination)
         {
             Debug.Assert(destination.Length >= _hashSizeInBytes);
 
-            byte[] srcArray = Array.Empty<byte>();
-            int srcLength = 0;
-            if (_buffer != null)
-            {
-                srcArray = _buffer.GetBuffer();
-                srcLength = (int)_buffer.Length;
-            }
+            ReadOnlySpan<byte> source = _buffer != null ?
+                new ReadOnlySpan<byte>(_buffer.GetBuffer(), 0, (int)_buffer.Length) :
+                default;
 
-            unsafe
-            {
-                fixed (byte* key = _key)
-                fixed (byte* src = srcArray)
-                fixed (byte* dest = destination)
-                {
-                    int res = Interop.BrowserCrypto.Sign(_hashAlgorithm, key, _key.Length, src, srcLength, dest, destination.Length);
-                    Debug.Assert(res != 0);
-                }
-            }
+            Sign(_hashAlgorithm, _key, source, destination);
 
             return _hashSizeInBytes;
         }
 
-        public static unsafe int MacDataOneShot(string hashAlgorithmId, ReadOnlySpan<byte> key, ReadOnlySpan<byte> data, Span<byte> destination)
+        public static int MacDataOneShot(string hashAlgorithmId, ReadOnlySpan<byte> key, ReadOnlySpan<byte> data, Span<byte> destination)
         {
             (SimpleDigest hashName, int hashSizeInBytes) = SHANativeHashProvider.HashAlgorithmToPal(hashAlgorithmId);
             Debug.Assert(destination.Length >= hashSizeInBytes);
 
+            Sign(hashName, key, data, destination);
+
+            return hashSizeInBytes;
+        }
+
+        private static unsafe void Sign(SimpleDigest hashName, ReadOnlySpan<byte> key, ReadOnlySpan<byte> data, Span<byte> destination)
```

Comment:
There's a philosophical debate as to whether or not a MAC is a Signature (basically, the question is is it a hierarchy, like `abstract concept Signature {}`, `abstract concept MAC : Signature {}`, `abstract concept AsymmetricSignature : Signature {}`, or if Signature is just synonymous with Asymmetric Signature.

NIST's glossary says both: https://csrc.nist.gov/glossary/term/digital_signature  (definition 1 is very general, definition 2 limits it to asymmetric).

`Mac`/`MacData` would be better names.  Though it looks like the interop call got named Sign, so, eh, whatever :smile:

Reply:
> Though it looks like the interop call got named Sign

Yeah, I named it that to align with the SubtleCrypto method name: https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/sign